doc/example-keywords.py

```diff
@@ -0,0 +1,211 @@
+        EXIT_CODE = 1
+        logging.warning(f"No keywords found in {filename}")
+        return False
+    keywords = set(kw.strip() for kw in match[1].split(","))
+    return keywords
+
+
+def get_matlab_keywords(filename):
+    """
+    Get keywords from the first comment block of a Matlab example. Keywords start after
+    a line starting with "Keywords:" and end with either the end of  the comment block
+    or with a line only containing the comment character (%). Individual keywords are
+    comma separated.
+    """
+    text = Path(filename).read_text()
+    match = re.search(r"(?:%.*?\n)+", text, re.DOTALL | re.MULTILINE)
+    global EXIT_CODE
+    if not match:
+        EXIT_CODE = 1
+        logging.error(f"Couldn't parse docstring for {filename}")
+        return False
+    docstring = match.group(0) + "\n\n"
+    docstring = "\n".join(line.lstrip("% ") for line in docstring.split("\n"))
+
+    match = re.search(r"\s*Keywords:(.*?)\n\n", docstring, re.DOTALL | re.MULTILINE)
+    if not match:
+        EXIT_CODE = 1
+        logging.warning(f"No keywords found in {filename}")
+        return False
+    keywords = set(kw.strip() for kw in match[1].split(","))
+    return keywords
+
+
+def get_cxx_keywords(filename):
+    text = Path(filename).read_text()
+    match = re.search(r"\/\*[!\*](.*?)\*\/", text, re.DOTALL | re.MULTILINE)
+    global EXIT_CODE
+    if not match:
+        EXIT_CODE = 1
+        logging.error(f"Couldn't parse docstring for {filename}")
+        return False
+    docstring = match.group(1) + "\n\n"
+    docstring = "\n".join(line.lstrip("* ") for line in docstring.split("\n"))
+    match = re.search(r"\s*Keywords:(.*?)\n\n", docstring, re.DOTALL | re.MULTILINE)
+    if not match:
+        EXIT_CODE = 1
+        logging.warning(f"No keywords found in {filename}")
+        return False
+    keywords = set(kw.strip() for kw in match[1].split(","))
+    return keywords
+
+
+def get_fortran_keywords(filename, comment_char):
+    text = Path(filename).read_text()
+    match = re.search(fr"(?:{comment_char}.*?\n)+", text,
+                      re.DOTALL | re.MULTILINE | re.IGNORECASE)
+    global EXIT_CODE
+    if not match:
+        EXIT_CODE = 1
+        logging.error(f"Couldn't parse docstring for {filename}")
+        return False
+    docstring = match.group(0) + "\n\n"
+    docstring = "\n".join(line.lstrip(f"{comment_char} ")
+                          for line in docstring.split("\n"))
+
+    match = re.search(r"\s*Keywords:(.*?)\n\n", docstring, re.DOTALL | re.MULTILINE)
+    if not match:
+        EXIT_CODE = 1
+        logging.warning(f"No keywords found in {filename}")
+        return False
+    keywords = set(kw.strip() for kw in match[1].split(","))
+    return keywords
+
+
+def get_all_keywords():
+    """
+    Read keywords from all Cantera examples and print out a summary list
+    """
+    text = (Path(__file__).parent / "example-skip-keywords.txt").read_text()
+    skip = set(text.split("\n"))
```

Comment:
```suggestion
    skip = set(text.splitlines())
```

This will handle `\r\n` as well

Reply:
The file is being read in the "universal newlines" mode, so `\r\n` is going to be automatically translated into `\n` no matter what. But I can change this regardless.